app/Cardano/Shell/Features/Logging.hs

```diff
@@ -60,33 +42,36 @@ testRotationParameters = RotationParameters
 -- the functions effects and constraining the user (programmer) of those function to use specific effects in them.
 -- https://github.com/input-output-hk/cardano-sl/blob/develop/util/src/Pos/Util/Log/LogSafe.hs
 data LoggingLayer = LoggingLayer
-    { iolLogDebug :: forall m. (MonadIO m) => Text -> m ()
-    , iolLogInfo  :: forall m. (MonadIO m) => Text -> m ()
-    , iolNonIo    :: forall m. (MonadThrow m) => m ()
-    }
-
-testLoggingLayer :: LoggingLayer
-testLoggingLayer = LoggingLayer
-    { iolLogDebug               = liftIO . putTextLn
-    , iolLogInfo                = liftIO . putTextLn
-    , iolNonIo                  = pure ()
+    { llStartTrace  :: forall m. (MonadIO m) => Trace m
+    , llLogDebug    :: forall m. (MonadIO m) => Trace m -> Text -> m ()
+    , llLogInfo     :: forall m. (MonadIO m) => Trace m -> Text -> m ()
+    , llLogNotice   :: forall m. (MonadIO m) => Trace m -> Text -> m ()
+    , llLogWarning  :: forall m. (MonadIO m) => Trace m -> Text -> m ()
+    , llLogError    :: forall m. (MonadIO m) => Trace m -> Text -> m ()
+    , llAppendName  :: forall m. (MonadIO m) => Text -> Trace m -> m (Trace m)
     }
 
 --------------------------------
 -- Feature
 --------------------------------
 
-type LoggingCardanoFeature = CardanoFeatureInit NoDependency RotationParameters LoggingLayer
+type LoggingCardanoFeature = CardanoFeatureInit NoDependency LoggingParameters LoggingLayer
 
 createLoggingFeature :: CardanoEnvironment -> CardanoConfiguration -> IO (LoggingLayer, CardanoFeature)
 createLoggingFeature cardanoEnvironment cardanoConfiguration = do
     -- we parse any additional configuration if there is any
     -- We don't know where the user wants to fetch the additional configuration from, it could be from
     -- the filesystem, so we give him the most flexible/powerful context, @IO@.
-    loggingConfiguration    <-  pure testRotationParameters
+    loggingConfiguration    <-  pure $ LoggingParameters
```

Comment:
The idea was to parse the configuration here.
So `CardanoConfiguration` should contain only the configuration related to _all features_ and any specific configurations should be loaded here.
For example, you can add a parser here and load some configuration from disk?